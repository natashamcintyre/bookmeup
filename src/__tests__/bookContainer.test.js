import React from 'react'
import BookContainer from '../components/bookContainer'

describe('bookContainer', () => {
  it('renders without crashing', () => {
    const book = JSON.stringify({ title: 'test_title', authors: [{ name: 'test_author' }], identifiers: { isbn_13: 'test_isb' }, cover: { medium: 'test_url' } })
    const users = [{ username: 'test_username', email: 'test_email', location: 'test_postcode' }]
    const data = { book: book, users: users }
    const component = shallow(<BookContainer className='book' data={data}/>)
    expect(component).toMatchSnapshot()
  })

  it('takes books as props and displays them', () => {
    const book = JSON.stringify({ title: 'test_title', authors: [{ name: 'test_author' }], identifiers: { isbn_13: 'test_isb' }, cover: { medium: 'test_url' } })
    const users = [{ username: 'test_username', email: 'test_email', location: 'test_postcode' }]
    const data = { book: book, users: users }
    const component = shallow(<BookContainer className='book' data={data}/>)
    expect(component.find('div.inner-book-container').children().length).toBe(4)
  })

  it('has a button to show more info', () => {
    const showModalMock = jest.fn()
    const book = JSON.stringify({ title: 'test_title', authors: [{ name: 'test_author' }], identifiers: { isbn_13: 'test_isb' }, cover: { medium: 'test_url' } })
    const users = [{ username: 'test_username', email: 'test_email', location: 'test_postcode' }]
    const data = { book: book, users: users }
    const component = shallow(<BookContainer className='book' data={data} showModal={showModalMock}/>)
    component.find('button').simulate('click')
    expect(showModalMock.mock.calls.length).toBe(1)
  })
})
