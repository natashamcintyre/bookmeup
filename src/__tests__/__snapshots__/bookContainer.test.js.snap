// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookContainer renders without crashing 1`] = `ShallowWrapper {}`;
